File: urbanairship-sdk/src/test/java/com/urbanairship/location/CircularRegionTest.java
/* Copyright 2017 Urban Airship and Contributors */

package com.urbanairship.location;

import com.urbanairship.BaseTestCase;

import org.junit.Test;

import static junit.framework.Assert.assertFalse;
import static junit.framework.Assert.assertTrue;

public class CircularRegionTest extends BaseTestCase {

    /**
     * Test isValid returns true for valid circular region.
     */
    @Test
    public void testValidCircularRegion() {
        CircularRegion circularRegion = new CircularRegion(10, 0.0, 0.0);
        assertTrue(circularRegion.isValid());
    }

    /**
     * Test setting radius above the max allowed is invalid.
     */
    @Test
    public void testRadiusAboveMax() {
        CircularRegion circularRegion = new CircularRegion(100001, 0.0, 0.0);
        assertFalse(circularRegion.isValid());
    }

    /**
     * Test setting radius below the min allowed is invalid.
     */
    @Test
    public void testRadiusBelowMin() {
        CircularRegion circularRegion = new CircularRegion(-1, 0.0, 0.0);
        assertFalse(circularRegion.isValid());
    }

    /**
     * Test setting a latitude above the max allowed is invalid.
     */
    @Test
    public void testLatitudeAboveMax() {
        CircularRegion circularRegion = new CircularRegion(10, 91.0, 0.0);
        assertFalse(circularRegion.isValid());
    }

    /**
     * Test setting latitude below the min allowed is invalid.
     */
    @Test
    public void testLatitudeBelowMin() {
        CircularRegion circularRegion = new CircularRegion(10, -91.0, 0.0);
        assertFalse(circularRegion.isValid());
    }

    /**
     * Test setting a longitude above the max allowed is invalid.
     */
    @Test
    public void testLongitudeAboveMax() {
        CircularRegion circularRegion = new CircularRegion(10, 0.0, 181.0);
        assertFalse(circularRegion.isValid());
    }

    /**
     * Test setting longitude below the min allowed is invalid.
     */
    @Test
    public void testLongitudeBelowMin() {
        CircularRegion circularRegion = new CircularRegion(10, 0.0, -181.0);
        assertFalse(circularRegion.isValid());
    }
}

File: urbanairship-core/src/test/java/com/urbanairship/analytics/location/CircularRegionTest.kt
/* Copyright Airship and Contributors */
package com.urbanairship.analytics.location

import androidx.test.ext.junit.runners.AndroidJUnit4
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
public class CircularRegionTest {

    /**
     * Test isValid returns true for valid circular region.
     */
    @Test
    public fun testValidCircularRegion() {
        val circularRegion = CircularRegion(10.0, 0.0, 0.0)
        assertTrue(circularRegion.isValid)
    }

    /**
     * Test setting radius above the max allowed is invalid.
     */
    @Test
    public fun testRadiusAboveMax() {
        val circularRegion = CircularRegion(100001.0, 0.0, 0.0)
        assertFalse(circularRegion.isValid)
    }

    /**
     * Test setting radius below the min allowed is invalid.
     */
    @Test
    public fun testRadiusBelowMin() {
        val circularRegion = CircularRegion(-1.0, 0.0, 0.0)
        assertFalse(circularRegion.isValid)
    }

    /**
     * Test setting a latitude above the max allowed is invalid.
     */
    @Test
    public fun testLatitudeAboveMax() {
        val circularRegion = CircularRegion(10.0, 91.0, 0.0)
        assertFalse(circularRegion.isValid)
    }

    /**
     * Test setting latitude below the min allowed is invalid.
     */
    @Test
    public fun testLatitudeBelowMin() {
        val circularRegion = CircularRegion(10.0, -91.0, 0.0)
        assertFalse(circularRegion.isValid)
    }

    /**
     * Test setting a longitude above the max allowed is invalid.
     */
    @Test
    public fun testLongitudeAboveMax() {
        val circularRegion = CircularRegion(10.0, 0.0, 181.0)
        assertFalse(circularRegion.isValid)
    }

    /**
     * Test setting longitude below the min allowed is invalid.
     */
    @Test
    public fun testLongitudeBelowMin() {
        val circularRegion = CircularRegion(10.0, 0.0, -181.0)
        assertFalse(circularRegion.isValid)
    }
}
